mart/nn/nn.py

```diff
@@ -94,15 +94,29 @@ def parse_sequence(self, sequence):
 
     def forward(self, step=None, **kwargs):
         # Try to fetch the customized architectural graph.
-        sequence = self._sequences[step]
+        if "sequence" not in kwargs:
+            # Backward compatible. We may get rid of step in the future.
+            sequence = self._sequences[step]
+        else:
+            sequence = kwargs["sequence"]
+
+        # Make a copy of sequence, because it will be destructed in the while loop.
+        sequence = sequence.copy()
+
+        while len(sequence) > 0:
+            # Don't pop the first element yet, because it may be used to re-evaluate the model.
+            key, module = next(iter(sequence.items()))
+            kwargs["sequence"] = sequence
```

Comment:
I don't like manipulating `kwargs` like this. It makes it hard to keep track of what is in `kwargs` as any given time. Instead, I would pass `sequence=sequence` where ever you need it.

Reply:
Fixed.